CMakeLists.txt

```diff
@@ -434,6 +435,28 @@ else()
   find_package(CUDAToolkit)
 endif()
 
+include(CMakeDependentOption)
+
+if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL "11.0" AND CMAKE_CUDA_COMPILER_ID MATCHES "NVIDIA" AND NOT QUDA_JITIFY)
```

Comment:
confused? why don't you use CMAKE_DEPENDENT_OPTION here?
That should make all the error if statements unnecessary

Reply:
It makes it more explicit to the user if they to try to enable it and it doesn't tell them that you can't.  So we have a working default for Jitify and clang cuda, but we don't secretly disable it if the user asks for it.  This seems like the right balance.

Reply:
So in that case`include(CMakeDependentOption)` is unnecessary. 
Can't we just put NVCC and CUDA>= 11 in the documentation
I lean towards CMAKE_DEPENDENT_OPTION to keep the CMakeLIsts shorter.